homeassistant/components/freebox/sensor.py

```diff
@@ -105,14 +104,16 @@ async def async_setup_entry(
         for description in DISK_PARTITION_SENSORS
     )
 
-    for node in router.home_devices.values():
-        for endpoint in node["show_endpoints"]:
-            if (
-                endpoint["name"] == "battery"
-                and endpoint["ep_type"] == "signal"
-                and endpoint.get("value") is not None
-            ):
-                entities.append(FreeboxBatterySensor(hass, router, node, endpoint))
+    entities.extend(
+        FreeboxBatterySensor(router, node, endpoint)
+        for node in router.home_devices.values()
+        for endpoint in node["show_endpoints"]
+        if (
+            endpoint["name"] == "battery"
+            and endpoint["ep_type"] == "signal"
+            and endpoint.get("value") is not None
+        )
+    )
```

Comment:
Apparently this is a bug in ruff, if you overwrite the type with something new it doesn't want you to use a list comprehension